kore/src/Kore/Builtin/Bool.hs

```diff
@@ -334,21 +358,24 @@ matchUnifyBoolNot first second
     | Just boolNot <- matchBoolNot first
       , isFunctionPattern first
       , Just value <- matchBool second =
-        Just $ UnifyBoolNot boolNot value
+        Just UnifyBoolNot{boolNot, value, term = second}
+    | Just boolNot <- matchBoolNot second
+      , isFunctionPattern second
+      , Just value <- matchBool first =
+        Just UnifyBoolNot{boolNot, value, term = first}
     | otherwise = Nothing
 {-# INLINE matchUnifyBoolNot #-}
 
 unifyBoolNot ::
     forall unifier.
     TermSimplifier RewritingVariableName unifier ->
-    TermLike RewritingVariableName ->
     UnifyBoolNot ->
     unifier (Pattern RewritingVariableName)
-unifyBoolNot unifyChildren term unifyData =
+unifyBoolNot unifyChildren unifyData =
     let notValue = asInternal (termLikeSort term) (not value)
      in unifyChildren notValue operand
   where
-    UnifyBoolNot{boolNot, value} = unifyData
+    UnifyBoolNot{boolNot, value, term} = unifyData
```

Comment:
Do we just use the `term` here to get the sort? Maybe we should save that in `UnifyBoolNot` instead. I think keeping the `Bool` value inside an `InternalBool` would give you all the information you need, please check if this is the case.